File: council/output.py
"""Concise decision output for stdout."""

from __future__ import annotations

from .types import Session


def format_decision_record(session: Session) -> str:
    """Concise executive summary — outcome, motion, votes."""
    outcome = session.outcome
    yays = [v for v in session.votes if v.vote == "yay"]
    nays = [v for v in session.votes if v.vote == "nay"]

    sections = [
        f"# Council Decision: {session.question}",
        f"**Outcome: {outcome.upper()}** ({len(yays)}-{len(nays)})",
    ]

    # Vote breakdown
    vote_lines = []
    for v in session.votes:
        icon = "Y" if v.vote == "yay" else "N"
        vote_lines.append(f"- [{icon}] **{v.agent.title()}**: {v.reason}")
    sections.append("## Votes\n\n" + "\n".join(vote_lines))

    # Key concerns from final round (informational only)
    max_round = max((t.round for t in session.turns), default=0)
    final_turns = [t for t in session.turns if t.round == max_round]
    all_concerns = []
    for turn in final_turns:
        for concern in turn.parsed.concerns:
            all_concerns.append(f"- **{turn.agent.title()}**: {concern}")
    if all_concerns:
        sections.append("## Outstanding Concerns\n\n" + "\n".join(all_concerns))

    return "\n\n".join(sections)

File: src/output.rs
use crate::orchestrator::title_case;
use crate::types::{Session, VoteChoice};

pub fn format_decision_record(session: &Session) -> String {
    let outcome = session.outcome();
    let yays: Vec<_> = session
        .votes
        .iter()
        .filter(|v| v.vote == VoteChoice::Yay)
        .collect();
    let nays: Vec<_> = session
        .votes
        .iter()
        .filter(|v| v.vote == VoteChoice::Nay)
        .collect();

    let mut sections = vec![
        format!("# Council Decision: {}", session.motion()),
    ];
    if session.crafted_motion.is_some() {
        sections.push(format!("*Original question: {}*", session.question));
    }
    sections.push(format!(
        "**Outcome: {}** ({}-{})",
        outcome.upper(),
        yays.len(),
        nays.len()
    ));

    // Vote breakdown
    let vote_lines: Vec<String> = session
        .votes
        .iter()
        .map(|v| {
            let icon = if v.vote == VoteChoice::Yay {
                "Y"
            } else {
                "N"
            };
            format!("- [{}] **{}**: {}", icon, title_case(&v.agent), v.reason)
        })
        .collect();
    sections.push(format!("## Votes\n\n{}", vote_lines.join("\n")));

    // Key concerns from final round
    let max_round = session.turns.iter().map(|t| t.round).max().unwrap_or(0);
    let final_turns: Vec<_> = session
        .turns
        .iter()
        .filter(|t| t.round == max_round)
        .collect();
    let mut all_concerns = Vec::new();
    for turn in final_turns {
        for concern in &turn.parsed.concerns {
            all_concerns.push(format!(
                "- **{}**: {}",
                title_case(&turn.agent),
                concern
            ));
        }
    }
    if !all_concerns.is_empty() {
        sections.push(format!(
            "## Outstanding Concerns\n\n{}",
            all_concerns.join("\n")
        ));
    }

    sections.join("\n\n")
}
